components/raftstore/src/store/fsm/store.rs

```diff
@@ -827,48 +852,57 @@ impl<EK: KvEngine, ER: RaftEngine, T: Transport> PollHandler<PeerFsm<EK, ER>, St
                 self.poll_ctx.raft_metrics.ready.message - self.previous_metrics.message,
                 self.poll_ctx.raft_metrics.ready.snapshot - self.previous_metrics.snapshot
             );
-        }
+            dur
+        } else {
+            now.saturating_duration_since(self.timer)
+        };
 
         self.poll_ctx.current_time = None;
         self.poll_ctx
             .raft_metrics
             .process_ready
             .observe(duration_to_sec(dur));
 
-        if now.saturating_duration_since(self.last_flush_time) >= Duration::from_millis(1) {
-            self.last_flush_time = now;
-            self.need_flush_events = false;
-            self.flush_events();
-        } else {
-            self.need_flush_events = true;
-        }
-
         if !self.poll_ctx.pending_latency_inspect.is_empty() {
+            let latency_inspect = std::mem::take(&mut self.poll_ctx.pending_latency_inspect);
             for inspector in &mut self.poll_ctx.pending_latency_inspect {
                 inspector.record_store_process(dur);
             }
```

Comment:
`self.poll_ctx.pending_latency_inspect` will be empty after `mem::take`

Reply:
Fixed